src/wrapMutator.ts

```diff
@@ -0,0 +1,21 @@
+import { action } from 'mobx';
+import ActionMessage from './interfaces/ActionMessage';
+import MutatorFunction from './interfaces/MutatorFunction';
+import { getGlobalContext } from './globalContext';
+
+// Wraps the target function for use as a mutator
+export default function wrapMutator<T extends ActionMessage>(
+    target: MutatorFunction<T>
+): MutatorFunction<T> {
+    // Wrap the target in a MobX action so it can modify the store
+    return action((actionMessage: T) => {
+        try {
+            getGlobalContext().inMutator = true;
+            if (target(actionMessage)) {
+                throw new Error('Mutators cannot return a value and cannot be async.');
```

Comment:
This seems to contradict the fact that you can return a value in a `createMutator().handles()` handler

Reply:
Yeah, I can see how this is confusing.  Effectively this will only ever apply to the existing mutators (which are functions), not the new-style mutators (which are objects that can have handler functions registered onto them).

Fortunately there should be no confusion to the consumer -- if they're using the new style mutators they'll never see this, and if they do see it with an old-style mutator the callstack will make it obvious where it's coming from.

---
In reply to: [200792183](https://github.com/Microsoft/satcheljs/pull/100#discussion_r200792183) [](ancestors = 200792183)